Batch mode (#76)
manekinekko/angular-web-bluetooth: README.md

}
```

## 2.c) subscribe to multiple services/characteristics on the same device 

> While waiting for a more native implementation, you can perform the batch operations mode my calling `discover$()` only once and then merge all streamed services/characteristics using RxJS as a workaround:

Reviewer: ```suggestion
You can perform the batch operations mode my calling `discover$()` only once and then merge all streamed services/characteristics using RxJS:
```

```javascript
import { Injectable } from '@angular/core';
import { EMPTY, merge, of } from 'rxjs';
import { catchError, map, mergeMap } from 'rxjs/operators';
import { BluetoothCore } from '@manekinekko/angular-web-bluetooth';

@Injectable({
  providedIn: 'root'
})
export class BleBatchService {
  private config;

  constructor(public ble: BluetoothCore) {

>     // 1) define supported services/characteristics that interest you 

Reviewer: ```suggestion
    // 1) define services/characteristics that you want to read data from 
```

    this.config = [{
      decoder: (value: DataView) => {...},
      service: 'service 1',
      characteristic: 'characteristic 2',
    }, {
      decoder: (value: DataView) => {...},
      service: 'service 2',
      characteristic: 'characteristic 2',
    }, ...];
  }
  
  values() {
    // 2) call discover$ to trigger the discovery process
    this.ble.discover$({
      acceptAllDevices: true,
      optionalServices: config.map(c => c.service)
    }).subscribe(...);
  

>     // 3) for every service/characteristic couple

Reviewer: ```suggestion
    // 3) for every service or characteristic pair
```

              // 7) same for eventual characteristic errors
              .pipe(catchError(error => {
                // your custom characteristic errors handling
                return EMPTY;
              }))),
            // 7) read characteristic value
            mergeMap((gattCharacteristic: BluetoothRemoteGATTCharacteristic) => this.ble.readValue$(gattCharacteristic)),
            // 8) decode characteristic value
            map((dataView: DataView) => c.decoder(dataView)),
          )
      });
    });
  
    // 9) combine the result array of values streams as a single stream
    return merge(...values);
  }
}
```

Please check the full implementation of `ble-batch.service.ts` [here](./src/app/ble-batch.service.ts) and also this demo component [batch-mode.component.ts](./src/app/batch-mode/batch-mode.component.ts) if you're looking for an example of how the service can be used.

## API documentation

The API documentation can be found here:  https://manekinekko.github.io/angular-web-bluetooth/

## Need a starter?

<img src="src/assets/screenshot.png"/>

This project serves also as a starter. Run the following command:

```bash
npm start
```

## Blog post

> Checkout my full [blog post on dev.to](https://dev.to/angular/the-web-bluetooth-module-for-angular-314b) about how to use this package in your app.

Reviewer: ```suggestion
Checkout the full [blog post on dev.to](https://dev.to/angular/the-web-bluetooth-module-for-angular-314b) about how to use this package in your app.
```

## Have a PR?

> All contributions are welcome. Here are few [open issues](https://github.com/manekinekko/angular-web-bluetooth/issues?q=is%3Aissue+is%3Aopen+label%3A%22help+wanted%22) that I need help with ;)

Reviewer: ```suggestion
All contributions are welcome. Here are few [open issues](https://github.com/manekinekko/angular-web-bluetooth/issues?q=is%3Aissue+is%3Aopen+label%3A%22help+wanted%22) that we need help with ;)
```
